File: problems_026to050/problem_034/solution.py
import os
import sys
from math import factorial

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0

        for num in range(10, 2540161):
            sum_of_factorials = sum(factorial(int(digit)) for digit in str(num))

            if sum_of_factorials == num:
                answer += num

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_034/solution.js
const { sum } = require('mathjs');

const { factorial } = require('../../lib/toolbox');

const solution = () => {
  let answer = 0;

  const factorialPerDigit = Array(10).fill(null).map((_, index) => Number(factorial(index)));

  for (let num = 10; num <= 2540160; num += 1) {
    const sumOfFactorials = sum(String(num).split('').map((digit) => factorialPerDigit[digit]));

    if (sumOfFactorials === num) {
      answer += num;
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
